src/components/button/createButton.ts

```diff
@@ -2,33 +2,64 @@ import { ComposeFactory } from 'dojo-compose/compose';
 import { VNodeProperties } from 'dojo-interfaces/vdom';
 import createWidgetBase from '../../createWidgetBase';
 import { Widget, WidgetOptions, WidgetProperties, WidgetState } from './../../interfaces';
-import createFormFieldMixin, { FormFieldMixin, FormFieldMixinState, FormFieldMixinOptions } from '../../mixins/createFormFieldMixin';
 
-export interface ButtonState extends WidgetState, FormFieldMixinState<string> {
-	label?: string;
+export interface ButtonState extends WidgetState {
+	/*
+	 * Text content of button
+	 */
+	content?: string;
+
+	/**
+	 * The button's name
+	 */
+	name?: string;
+
+	/**
+	 * Accessibility attributes
+	 */
+	disabled?: boolean;
+	pressed?: boolean;
 }
 
 export interface ButtonProperties extends WidgetProperties {
-	label?: string;
+	content?: string;
+	name?: string;
+	type?: string;
+
+	/**
+	 * Accessibility attributes
+	 */
+	descriptionID?: string;
+	disabled?: boolean;
+	role?: string;
+	pressed?: boolean;
+	hasPopup?: boolean;
 }
 
-export interface ButtonOptions extends WidgetOptions<ButtonState, ButtonProperties>, FormFieldMixinOptions<any, ButtonState> { }
+export interface ButtonOptions extends WidgetOptions<ButtonState, ButtonProperties> { }
 
-export type Button = Widget<ButtonState, ButtonProperties> & FormFieldMixin<string, ButtonState>;
+export type Button = Widget<ButtonState, ButtonProperties>;
 
 export interface ButtonFactory extends ComposeFactory<Button, ButtonOptions> { }
 
 const createButton: ButtonFactory = createWidgetBase
-	.mixin(createFormFieldMixin)
 	.mixin({
 		mixin: {
 			nodeAttributes: [
 				function(this: Button): VNodeProperties {
-					return { innerHTML: this.state.label };
+					return {
+						innerHTML: this.properties['content'],
+						type: this.properties['type'],
+						name: this.properties['name'],
+						role: this.properties['role'],
+						disabled: this.state.disabled,
+						'aria-pressed': this.state.pressed,
+						'aria-describedby': this.properties['descriptionID'],
+						'aria-haspopup': this.properties['hasPopup']
+					};
 				}
 			],
-			tagName: 'button',
-			type: 'button'
```

Comment:
There are different possible types, but I don't think it should change after creation (same reason as above). Having a type of `button` with the tagname `button` is redundant though, so I removed it as a default.